tests/queries/0_stateless/00970_substring_arg_validation.reference

```diff
@@ -1 +1 @@
-o
+
```

Comment:
Why this and 00498 tests output changed ? How this change affect backward compatibility ?

Reply:
Some old `substring` and `arraySlice` test. If the `offset` and `length` is negative constant  and `value` is not constant, result is Incorrect.

Reply:
For this case
```sql
SELECT substring(materialize('hello'), -1, -1);
```
Old
```text
┌─substring(materialize('hello'), -1, -1)─┐
│ o                                        │
└─────────────────────────────────────────┘
```
New
```text
┌─substring(materialize('hello'), -1, -1)─┐
│                                         │
└─────────────────────────────────────────┘
```